### PR TITLE
Pass context to arweave api

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/cron/arweave/mod.rs [hunk 1]
```diff
@@ -1,2 +1,254 @@
...
+                "http" => ArweaveProtocol::Http,
+                "https" | _ => ArweaveProtocol::Https,
+            },
+        }
+    }
+
+    pub async fn get_tx_data<Context>(
+        &self,
+        transaction_id: &str,
+        ctx: &Context,
```
Comment (mikonieminen) [inline src/cron/arweave/mod.rs]: Minor thing, but I would move context as the first arg (apart from self). Usually context or environment kind of args are passed before actual input args. Not really a big thing, but just common ordering what you can see also elsewhere.

File: src/cron/bundle.rs [hunk 2]
```diff
@@ -48,11 +49,11 @@ pub async fn validate_bundler<Context>(
     bundler: Bundler,
 ) -> Result<(), ValidatorCronError>
 where
-    Context: queries::RequestContext,
+    Context: queries::RequestContext + arweave::ArweaveContext,
 {
     let arweave = Arweave::new(80, String::from("arweave.net"), String::from("http"));
     let txs_req = arweave
-        .get_latest_transactions(&bundler.address, Some(50), None)
+        .get_latest_transactions::<Context>(&bundler.address, Some(50), None, ctx)
```
Comment (mikonieminen) [inline src/cron/bundle.rs]: There should be no need to have `::<Context>` in the call. It should be inferred through args.

File: src/server/routes/post_tx.rs [hunk 3]
```diff
@@ -98,6 +102,22 @@ fn deep_hash_body(body: &PostTxBody) -> Result<Bytes, ValidatorServerError> {
...
 
-async fn add_to_db(_body: &PostTxBody) -> Result<(), ValidatorServerError> {
-    Ok(())
+async fn add_to_db<Config>(ctx: &Config, body: &PostTxBody) -> Result<(), ValidatorServerError>
+where
+    Config: RequestContext + 'static,
+{
+    let tx = NewTransaction {
+        id: body.id.clone(),
+        epoch: 0,
```
Comment (mikonieminen) [inline src/server/routes/post_tx.rs]: Shouldn't this be the current epoch? You can get the value through context: `server::routes::sign::Config::current_epoch(&self) -> i64`

That reminds me, the return value of this function is wrong, it should be `u128`.

**Review by mikonieminen — CHANGES_REQUESTED** [review]

File: src/server/routes/post_tx.rs (hunk 3 above)
Comment (ghost) [reply src/server/routes/post_tx.rs]: We discussed about this, currently the epoch should be left as i64 for Sqlite support. No problem

Comment (mikonieminen) [reply src/server/routes/post_tx.rs]: Sorry, I was a bit unclear. You're correct, the database table should have `i64`, while the API I pointed out to, should have `u128`. The reason is that all domain logic should use correct types while the database layer should have the compromise. But the main point is that this value should not be set to zero.

Comment (mikonieminen) [reply src/server/routes/post_tx.rs]: I will fix the implementation of `current_epoch` function at some later point.

**Comment by mikonieminen** [comment]
Better. How the access to `current_epoch` is implemented is not perfect, but we can fix it separately. Context implementation and traits need some refactoring anyway so it can be done with other context changes.

**Pull request merged**